tarantool/connection.py

```diff
@@ -1751,3 +1765,563 @@ def _check_features(self):
+            crud_resp = self.call("crud.insert", space_name, values, opts)
+        except DatabaseError as e:
+            if e.code == 33:
+                e = wrapCrudUndefinedProcedureError(e)
+            raise e
+
+        if crud_resp[1] is not None:
+            raise CrudModuleError(None, CrudError(crud_resp[1]))
+
+        return CrudResult(crud_resp[0])
+
+    def crud_insert_object(self, space_name: str, values: Union[tuple, list], opts={}) -> CrudResult:
+        """
+        TODO: write docs
+        """
+
+        assert isinstance(space_name, str)
+        assert isinstance(values, dict)
+        assert isinstance(opts, dict)
+
+        try:
+            crud_resp = self.call("crud.insert_object", space_name, values, opts)
+        except DatabaseError as e:
+            if e.code == 33:
+                e = wrapCrudUndefinedProcedureError(e)
+            raise e    
+
+        if crud_resp[1] is not None:
+            raise CrudModuleError(None, CrudError(crud_resp[1]))
+
+        return CrudResult(crud_resp[0])
+
+    def crud_insert_many(self, space_name: str, values: Union[tuple, list], opts={}) -> CrudResult:
+        """
+        TODO: write docs
+        """
+
+        assert isinstance(space_name, str)
+        assert isinstance(values, (tuple, list))
+        assert isinstance(opts, dict)
+
+        try:
+            crud_resp = self.call("crud.insert_many", space_name, values, opts)
+        except DatabaseError as e:
+            if e.code == 33:
+                e = wrapCrudUndefinedProcedureError(e)
+            raise e
+
+        res = None
+        if crud_resp[0] is not None:
+            res = CrudResult(crud_resp[0])
+
+        if crud_resp[1] is not None:
+            errs = list()
+            for err in crud_resp[1]:
+                errs.append(CrudError(err))
+            raise CrudModuleError(res, errs)
+
+        return res
+
+    def crud_insert_object_many(self, space_name: str, values: Union[tuple, list], opts={}) -> CrudResult:
+        """
+        TODO: write docs
+        """
+
+        assert isinstance(space_name, str)
+        assert isinstance(values, (tuple, list))
+        assert isinstance(opts, dict)
+
+        try:
+            crud_resp = self.call("crud.insert_object_many", space_name, values, opts)
+        except DatabaseError as e:
+            if e.code == 33:
+                e = wrapCrudUndefinedProcedureError(e)
+            raise e
+
+        res = None
+        if crud_resp[0] is not None:
+            res = CrudResult(crud_resp[0])
+
+        if crud_resp[1] is not None:
+            errs = list()
+            for err in crud_resp[1]:
+                errs.append(CrudError(err))
+            raise CrudModuleError(res, errs)
+
+        return res
+
+    def crud_get(self, space_name: str, key: int, opts={}) -> CrudResult:
+        """
+        TODO: write docs
+        """
+
+        assert isinstance(space_name, str)
+        assert isinstance(opts, dict)
+
+        try:
```

Comment:
It seems like you can extract this block to a single wrapper.

Reply:
Done, implemented `call_crud` in tarantool.crud